nixos/tests/docker-tools.nix

```
@@ -431,5 +431,58 @@ import ./make-test-python.nix ({ pkgs, ... }: {
         docker.succeed("docker run --rm image-with-certs:latest test -r /etc/pki/tls/certs/ca-bundle.crt")
         docker.succeed("docker image rm image-with-certs:latest")
 
+    with subtest("buildNixShellImage: Can build a basic derivation"):
+        docker.succeed(
+            "docker load --input='${examples.nix-shell-basic}'",
```

```suggestion
            "${examples.nix-shell-basic} | docker load",
```

See suggestions about `streamLayeredImage`.